test-framework/junit5/src/main/java/io/quarkus/test/junit/util/QuarkusTestProfileAwareClassOrderer.java

```diff
@@ -0,0 +1,87 @@
+package io.quarkus.test.junit.util;
+
+import java.util.Comparator;
+import java.util.Optional;
+
+import org.junit.jupiter.api.ClassDescriptor;
+import org.junit.jupiter.api.ClassOrderer;
+import org.junit.jupiter.api.ClassOrdererContext;
+
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+
+/**
+ * A {@link ClassOrderer} that orders {@link QuarkusTest} and {@link QuarkusIntegrationTest} classes for minimum Quarkus
+ * restarts by grouping them by their {@link TestProfile}.
+ * <p/>
+ * By default, Quarkus*Tests not using any profile come first, then classes using a profile (in groups) and then all other
+ * non-Quarkus tests (e.g. plain unit tests).
+ * <p/>
+ * Internally, ordering is based on three prefixes that are prepended to the fully qualified name of the respective class, with
+ * the fully qualified class name of the {@link io.quarkus.test.junit.QuarkusTestProfile QuarkusTestProfile} as an infix (if
+ * present).
+ * The default prefixes are defined by {@code DEFAULT_ORDER_PREFIX_*} and can be overridden in {@code junit-platform.properties}
+ * via {@code CFGKEY_ORDER_PREFIX_*}, e.g. non-Quarkus tests can be run first (not last) by setting
+ * {@link #CFGKEY_ORDER_PREFIX_NON_QUARKUS_TEST} to {@code 10_}.
+ * <p/>
+ * {@link #getCustomOrderKey(ClassDescriptor, ClassOrdererContext)} can be overridden to provide a custom order number for a
+ * given test class, e.g. based on {@link org.junit.jupiter.api.Tag}, class name or something else.
+ * <p/>
+ * Limitations:
+ * <ul>
+ * <li>This orderer does not (yet) consider {@linkplain io.quarkus.test.common.QuarkusTestResource#restrictToAnnotatedClass()
+ * test resources that are restricted to the annotated class}.</li>
+ * <li>Only JUnit5 test classes are subject to ordering, e.g. ArchUnit test classes are not passed to this orderer.</li>
+ * </ul>
+ */
+public class QuarkusTestProfileAwareClassOrderer implements ClassOrderer {
+
+    protected static final String DEFAULT_ORDER_PREFIX_QUARKUS_TEST = "20_";
+    protected static final String DEFAULT_ORDER_PREFIX_QUARKUS_TEST_WITH_PROFILE = "40_";
+    protected static final String DEFAULT_ORDER_PREFIX_NON_QUARKUS_TEST = "60_";
```

Comment:
Note: The "number spacing" might seem odd, but I wanted to leave some gaps for users to fill in their own orderings.